modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/DatabaseNodeService.java

```diff
@@ -464,15 +464,19 @@ void retrieveDatabase(
                     // so it is ok if this happens in a blocking manner on a thread from generic thread pool.
                     // This makes the code easier to understand and maintain.
                     SearchResponse searchResponse = client.search(searchRequest).actionGet();
-                    SearchHit[] hits = searchResponse.getHits().getHits();
+                    try {
+                        SearchHit[] hits = searchResponse.getHits().getHits();
 
-                    if (searchResponse.getHits().getHits().length == 0) {
-                        failureHandler.accept(new ResourceNotFoundException("chunk document with id [" + id + "] not found"));
-                        return;
+                        if (searchResponse.getHits().getHits().length == 0) {
+                            failureHandler.accept(new ResourceNotFoundException("chunk document with id [" + id + "] not found"));
+                            return;
+                        }
+                        byte[] data = (byte[]) hits[0].getSourceAsMap().get("data");
+                        md.update(data);
+                        chunkConsumer.accept(data);
+                    } finally {
+                        searchResponse.decRef();
```

Comment:
❓ (Not related to this PR)  Would it make sense to make `SearchResponse` implement `Autocloseable`, so clients of the class can decide to use try-with-resources or `RefCounted` for their purposes? 🤔

Reply:
Maybe :) We had this discussion before. It would certainly save some code, I'm not 100% sure about it because I always found it a bit weird to have the duplication between ref counting and closing that we have in `ReleasableBytesReference` but it's definitely something to look into still :)